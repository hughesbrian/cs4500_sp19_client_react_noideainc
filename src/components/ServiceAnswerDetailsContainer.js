import React from 'react'
import ServiceAnswerService from '../services/ServiceAnswerService'
import ServiceQuestionService from '../services/ServiceQuestionService'
import ServiceAnswerDetails from './ServiceAnswerDetails'
class ServiceAnswerDetailsContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceAnswerService = ServiceAnswerService.getInstance();
        this.serviceQuestionService = ServiceQuestionService.getInstance();
        this.state = {
            serviceAnswers: [],
            answer: {
                choiceAnswer: '',
                id: 0,
                trueFalseAnswer: '',
                minRangeAnswer: '',
                maxRangeAnswer: ''
            },
            createAnswer:{
                choiceAnswer: '',
                id: 0,
                trueFalseAnswer: '',
                minRangeAnswer: '',
                maxRangeAnswer: ''
            },

            serviceQuestions : [],
            serviceQuestion: {
                question: '',
                id: 1
            }
        }
    }
    componentDidMount() {
        this.findAllServiceAnswers();
        this.serviceQuestionService
            .findAllServiceQuestions()
            .then(serviceQuestions => {
            this.setState({
                serviceQuestions: serviceQuestions,
                serviceQuestion: serviceQuestions[0]
            })
        }
    )
    }
    findAllServiceAnswers = () =>
    this.serviceAnswerService
        .findAllServiceAnswers()
        .then(serviceAnswers => {
            //this.props.history.push("/admin/service-answers/" + serviceAnswers[0].id);
            this.setState({
                      serviceAnswers: serviceAnswers,
                      answer: serviceAnswers[0]
                  })
}
)
selectServiceAnswer = id =>
this.serviceAnswerService
    .findServiceAnswerById(id)
    .then(answer => {
    //this.props.history.push("/admin/service-answers/" + id);
    this.setState({
        answer: answer
    })
}
)

updateMinValue = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: e.target.value,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateChoiceAnswer = e =>
this.setState({
    answer: {
        choiceAnswer: e.target.value,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateMaxValue = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: e.target.value,
        id: this.state.answer.id
    }
})

updateTrueFalse = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: e.target.value,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewMinValue = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: e.target.value,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewChoiceAnswer = e =>
this.setState({
    createAnswer: {
        choiceAnswer: e.target.value,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewMaxValue = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: e.target.value,
        id: this.state.answer.id
    }
})

updateNewTrueFalse = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: e.target.value,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})



selectServiceQuestion = id =>
this.serviceQuestionService
    .findServiceQuestionById(id)
    .then(serviceQuestion => {
    this.setState({
        serviceQuestion: serviceQuestion
    })
}
)

createServiceAnswer = () =>
this.serviceAnswerService
    .createServiceAnswer(this.state.createAnswer)
    .then(this.findAllServiceAnswers());
deleteServiceAnswer = id =>
this.serviceAnswerService
    .deleteServiceAnswer(id)
    .then(this.findAllServiceAnswers());
updateServiceAnswer = () =>
this.serviceAnswerService.updateServiceAnswer(this.state.answer);

render = () => <ServiceAnswerDetails serviceAnswers = {this.state.serviceAnswers}
                answer = {this.state.answer}
                createAnswer = {this.state.createAnswer}
                serviceQuestion = {this.state.serviceQuestion}
                serviceQuestions = {this.state.serviceQuestions}
                findAllServiceAnswer = {this.findAllServiceAnswers}
                selectServiceAnswer = {this.selectServiceAnswer}
                updateMinValue = {this.updateMinValue}
                updateChoiceAnswer = {this.updateChoiceAnswer}
                updateMaxValue = {this.updateMaxValue}
                updateTrueFalse = {this.updateTrueFalse}
                updateNewMinValue = {this.updateNewMinValue}
                updateNewChoiceAnswer = {this.updateNewChoiceAnswer}
                updateNewMaxValue = {this.updateNewMaxValue}
                updateNewTrueFalse = {this.updateNewTrueFalse}
                selectServiceQuestion = {this.selectServiceQuestion}
                createServiceAnswer = {this.createServiceAnswer}
                deleteServiceAnswer = {this.deleteServiceAnswer}
                updateServiceAnswer = {this.updateServiceAnswer} />;
}

export default ServiceAnswerDetailsContainer