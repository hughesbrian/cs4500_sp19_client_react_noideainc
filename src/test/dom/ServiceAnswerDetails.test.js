import React from 'react'
import ServiceAnswerDetailsContainer from '../../components/ServiceAnswerDetailsContainer'
import TestRenderer from 'react-test-renderer'
import serviceAnswers from '../MockData/ServiceAnswer.mock.json'
import sleep from '../util/sleep'
import '../mockservice/ServiceAnswerService.mock.js'
import serviceCategories from "../MockData/ServiceCategories.mock";

test('ServiceAnswerDetails renders correctly', async () => {
    const testRenderer = TestRenderer.create(
        <ServiceAnswerDetailsContainer/>
    )
    await sleep(1000) // sleep to give the component a chance to render.

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root

    const select_answer = testInstance.findAllByProps({className: 'select-answer'})
    console.log(select_answer[0].props)
    await sleep(500) // sleep to give the component a chance to render.
    select_answer[0].props.onChange({target:{value: 1}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const select_question = testInstance.findByProps({className: 'select-question'})
    console.log(select_question.props)
    await sleep(500) // sleep to give the component a chance to render.
    select_question.props.onChange({target: {value: 1}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    /*const updateBtn = testInstance.findByProps({className: 'btn btn-success update-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    updateBtn.props.onClick()
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    await sleep(500) // sleep to give the component a chance to render.
    titleField.props.onChange({target: {value: "Creation"}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const createBtn = testInstance.findByProps({className: 'btn btn-primary create-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    createBtn.props.onClick()
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const deleteBtns = testInstance.findAllByProps({className: 'btn btn-danger delete-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    const last = deleteBtns.length - 1
    deleteBtns[last].props.onClick(serviceCategories[last].id)
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()*/

})
