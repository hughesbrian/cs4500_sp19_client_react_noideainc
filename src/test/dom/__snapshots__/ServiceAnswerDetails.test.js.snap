// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceAnswerDetails renders correctly 1`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <select
    className="select-answer"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={4}
    >
      4
    </option>
    <option
      value={5}
    >
      5
    </option>
    <option
      value={6}
    >
      6
    </option>
  </select>
  <label>
    Answer ID
  </label>
  <br />
  <input
    className="answer-id"
    onChange={[Function]}
    value={1}
  />
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="answer-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="answer-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="answer-choice"
    onChange={[Function]}
    value={0}
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="answer-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    onClick={[Function]}
  >
    Update
  </button>
  <h3>
    Service Question Details
  </h3>
  <select
    className="select-question"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      
    </option>
    <option
      value={2}
    >
      
    </option>
    <option
      value={3}
    >
      
    </option>
  </select>
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="create-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="create-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="create-choice"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="create-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create
  </button>
</div>
`;

exports[`ServiceAnswerDetails renders correctly 2`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <select
    className="select-answer"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={4}
    >
      4
    </option>
    <option
      value={5}
    >
      5
    </option>
    <option
      value={6}
    >
      6
    </option>
  </select>
  <label>
    Answer ID
  </label>
  <br />
  <input
    className="answer-id"
    onChange={[Function]}
    value={1}
  />
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="answer-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="answer-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="answer-choice"
    onChange={[Function]}
    value={0}
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="answer-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    onClick={[Function]}
  >
    Update
  </button>
  <h3>
    Service Question Details
  </h3>
  <select
    className="select-question"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      
    </option>
    <option
      value={2}
    >
      
    </option>
    <option
      value={3}
    >
      
    </option>
  </select>
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="create-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="create-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="create-choice"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="create-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create
  </button>
</div>
`;

exports[`ServiceAnswerDetails renders correctly 3`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <select
    className="select-answer"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={4}
    >
      4
    </option>
    <option
      value={5}
    >
      5
    </option>
    <option
      value={6}
    >
      6
    </option>
  </select>
  <label>
    Answer ID
  </label>
  <br />
  <input
    className="answer-id"
    onChange={[Function]}
    value={1}
  />
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="answer-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="answer-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="answer-choice"
    onChange={[Function]}
    value={0}
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="answer-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    onClick={[Function]}
  >
    Update
  </button>
  <h3>
    Service Question Details
  </h3>
  <select
    className="select-question"
    onChange={[Function]}
    value={1}
  >
    <option
      value={1}
    >
      
    </option>
    <option
      value={2}
    >
      
    </option>
    <option
      value={3}
    >
      
    </option>
  </select>
  <label>
    Answer minAnswer
  </label>
  <br />
  <input
    className="create-min"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer maxAnswer
  </label>
  <br />
  <input
    className="create-max"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer choiceAnswer
  </label>
  <br />
  <input
    className="create-choice"
    onChange={[Function]}
    value=""
  />
  <label>
    Answer trueFalseAnswer
  </label>
  <br />
  <input
    className="create-true-false"
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create
  </button>
</div>
`;
